test/jdk/java/lang/invoke/indify/Indify.java

```diff
@@ -242,62 +350,72 @@ public void indify(String a) throws IOException {
         File f = new File(a);
         String fn = f.getName();
         if (fn.endsWith(".class") && f.isFile())
-            indifyFile(f, dest);
+            indifyFile(f);
         else if (fn.endsWith(".jar") && f.isFile())
-            indifyJar(f, dest);
+            indifyJar(); //Not yet implemented
         else if (f.isDirectory())
-            indifyTree(f, dest);
+            indifyTree(f);
         else if (!keepgoing)
             throw new RuntimeException("unrecognized file: "+a);
     }
 
     private void ensureDirectory(File dir) {
         if (dir.mkdirs() && !quiet)
-            System.err.println("created "+dir);
+            System.err.println("Created new directory to: "+dir);
     }
 
-    public void indifyFile(File f, File dest) throws IOException {
-        if (verbose)  System.err.println("reading "+f);
-        ClassFile cf = new ClassFile(f);
-        Logic logic = new Logic(cf);
-        boolean changed = logic.transform();
+    public void indifyFile(File f) throws IOException {
+        if (verbose)  System.err.println("reading " + f);
+        ClassModel model = parseClassFile(f);
+        Logic logic = new Logic(model);
+        Boolean changed = logic.transform();
+        System.err.println("Class file transformation: " + changed);
         logic.reportPatternMethods(quiet, keepgoing);
-        if (changed || all) {
-            File outfile;
-            if (dest != null) {
+        writeNewClassFile(logic.classModel, changed, f);
+    }
+
+    void writeNewClassFile(ClassModel newClassModel, Boolean changed, File sourceFile) throws IOException {
+        byte[] new_bytes = transformToBytes(newClassModel);
+        if(changed || all){
```

Comment:
If you are using idea, select this whole method body and run ctrl+alt+L to format the code. A few improper formatting.